immuni_analytics/helpers/request.py

```diff
@@ -0,0 +1,77 @@
+#   Copyright (C) 2020 Presidenza del Consiglio dei Ministri.
+#   Please refer to the AUTHORS file for more information.
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU Affero General Public License as
+#   published by the Free Software Foundation, either version 3 of the
+#   License, or (at your option) any later version.
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#   GNU Affero General Public License for more details.
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+from typing import Any, Callable, Dict
+
+from aiohttp import ClientError, ClientResponse, ClientSession, ClientTimeout
+from tenacity import (
+    RetryCallState,
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_exponential,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ServerUnavailableError(Exception):
```

Comment:
IMHO, all exceptions within this project should inherit from ImmuniException.
Also, did we double-check none of these bubble-up to the API user? If so, we should inherit from the ApiException, with the message and error_code fields.

Reply:
They are raised in a celery task, ImmuniException will do.